cranelift/filetests/src/test_compile.rs

```diff
@@ -97,3 +114,74 @@ impl binemit::CodeSink for SizeSink {
     fn begin_rodata(&mut self) {}
     fn end_codegen(&mut self) {}
 }
+
+fn check_precise_output(text: &str, context: &Context) -> Result<()> {
+    let actual = text.lines().collect::<Vec<_>>();
+
+    // Use the comments after the function to build the test expectation.
+    let expected = context
+        .details
+        .comments
+        .iter()
+        .map(|c| c.text.strip_prefix("; ").unwrap_or(c.text))
+        .collect::<Vec<_>>();
+
+    // If the expectation matches what we got, then there's nothing to do.
+    if actual == expected {
+        return Ok(());
+    }
+
+    // If we're supposed to automatically update the test, then do so here.
+    if env::var("BLESS").unwrap_or(String::new()) == "1" {
```

Comment:
tiniest of nits: I think I'd prefer the environment variable to be namespaced a bit just to avoid an edge case where the user has an environment variable in their usual environment that causes accidental weirdness. (`BLESS` isn't out of the realm of possibility of some other tool also claiming...) Maybe `CRANELIFT_TEST_BLESS`?